test/generate-vdm.spec.ts

```diff
@@ -0,0 +1,144 @@
+
+      try {
+        await GenerateVdm.run(['-i', 'input', '-o', 'output', '--projectDir', getProjectDir()]);
+      } catch (e) {
+        expect(e.message).toContain('ENOENT: no such file or directory');
+        expect(spyToGeneratorSDK).toHaveReturnedWith(expected);
+      }
+    }
+  });
+
+  function getProjectDir() {
+    return path.resolve(__dirname, 'generate-vdm-spec');
+  }
+
+  function getInputAndExpected(key: keyof GeneratorOptionsSDK): { expected: GeneratorOptionsSDK; args: string[] } | undefined {
+    const args = ['-i', 'input', '-o', 'output', '--projectDir', getProjectDir()];
+    const expected = getDefault(getProjectDir());
+
+    const option = generatorOptionsSDK[key];
+    if (option && option.type === 'boolean') {
+      const casted = key as keyof GeneratorOptionsSDK;
+      args.push(`--${option.default ? 'no-' : ''}${key}`);
+      (expected[casted] as boolean) = !option.default;
+      return { args, expected };
+    }
+  }
+
+  it('should pass each boolean flags correctly', async () => {
+    // TODO there was a issue in the SDK for the service-mapping.json file and windows. Onve the fix is out put this test back int.
+    if (process.platform !== 'win32') {
+      for (const key of Object.keys(generatorOptionsSDK)) {
+        const argsExpected = getInputAndExpected(key as keyof GeneratorOptionsSDK);
+        if (argsExpected) {
+          try {
+            await GenerateVdm.run(argsExpected.args);
+          } catch (e) {
+            expect(e.message).toContain('ENOENT: no such file or directory');
+            expect(spyToGeneratorSDK).toHaveReturnedWith(argsExpected.expected);
+          }
+        }
+      }
+    }
+  }, 10000);
+
+  function getDefault(projectDir: string): GeneratorOptionsSDK {
+    return {
+      ...(Object.keys(generatorOptionsSDK).reduce((prev, current) => {
+        const value = generatorOptionsSDK[current as keyof GeneratorOptionsSDK];
+        if (value) {
+          prev[current as keyof GeneratorOptionsSDK] = value.default;
+        }
+        return prev;
+      }, {} as any) as GeneratorOptionsSDK),
+      inputDir: path.resolve(projectDir, 'input'),
+      outputDir: path.resolve(projectDir, 'output'),
+      serviceMapping: path.resolve(projectDir, 'input', 'service-mapping.json')
+    };
+  }
+
+  function getCliInputWithAllBooleanFlagsFalse(): string[] {
+    const allFalse = getParsedInputWithAllBooleanFlagsFalse();
+    const stringArguments = Object.entries(allFalse).reduce((collected: string[], [key, value]) => {
+      switch (typeof allFalse[key as keyof generateVdmUtil.FlagsParsed]) {
+        case 'string':
+          collected.push(`--${key}`, value!.toString());
+          break;
+        case 'boolean':
+          collected = generatorOptionsSDK[key as keyof GeneratorOptionsSDK]?.default ? [...collected, `--no-${key}`] : collected;
+          break;
+      }
+      return collected;
+    }, []);
+    return [...stringArguments];
```

Comment:
If you don't just return the result of the reduce above (see comment above), you can just straight return it here. The copying is unnecessary. 
```suggestion
    return stringArguments;
```